site/public/js/forum.js

```diff
@@ -1251,6 +1251,68 @@ function modifyThreadList(currentThreadId, currentCategoriesId, course, loadFirs
     });
 }
 
+function toggleLike(post_id,thread_id,current_user) {
+
+    // eslint-disable-next-line no-undef
+    const url = buildCourseUrl(['post', 'likes']);
+    $.ajax({
+        url: url,
+        type: 'POST',
+        data: {
+            post_id: post_id,
+            thread_id: thread_id,
+            current_user: current_user,
+            csrf_token: csrfToken,
+        },
+        success: function(data) {
+            try {
+                // eslint-disable-next-line no-var
+                var json = JSON.parse(data);
+            }
+            catch (err) {
+                // eslint-disable-next-line no-undef
+                displayErrorMessage('Error parsing data. Please try again.');
+                return;
+            }
+            if (json['status'] === 'fail') {
+                // eslint-disable-next-line no-undef
```

Comment:
You are ignoring quite a lot of eslint warnings. It would be better to fix these rather than ignore them.